servers/rendering/renderer_rd/forward_clustered/scene_shader_forward_clustered.cpp

```diff
@@ -250,8 +261,36 @@ void SceneShaderForwardClustered::ShaderData::set_code(const String &p_code) {
 
 	if (depth_test != DEPTH_TEST_DISABLED) {
 		depth_stencil_state.enable_depth_test = true;
-		depth_stencil_state.depth_compare_operator = RD::COMPARE_OP_LESS_OR_EQUAL;
 		depth_stencil_state.enable_depth_write = depth_draw != DEPTH_DRAW_DISABLED ? true : false;
+		switch (depth_function) {
+			case DEPTH_FUNCTION_LESS_OR_EQUAL:
+				depth_stencil_state.depth_compare_operator = RD::COMPARE_OP_LESS_OR_EQUAL;
+				break;
+			case DEPTH_FUNCTION_LESS:
+				depth_stencil_state.depth_compare_operator = RD::COMPARE_OP_LESS;
+				break;
+			case DEPTH_FUNCTION_GREATER_OR_EQUAL:
+				depth_stencil_state.depth_compare_operator = RD::COMPARE_OP_GREATER_OR_EQUAL;
+				break;
+			case DEPTH_FUNCTION_GREATER:
+				depth_stencil_state.depth_compare_operator = RD::COMPARE_OP_GREATER;
+				break;
+			case DEPTH_FUNCTION_EQUAL:
+				depth_stencil_state.depth_compare_operator = RD::COMPARE_OP_EQUAL;
+				break;
+			case DEPTH_FUNCTION_NOT_EQUAL:
+				depth_stencil_state.depth_compare_operator = RD::COMPARE_OP_NOT_EQUAL;
+				break;
+			case DEPTH_FUNCTION_ALWAYS:
+				depth_stencil_state.depth_compare_operator = RD::COMPARE_OP_ALWAYS;
+				break;
+			case DEPTH_FUNCTION_NEVER:
+				depth_stencil_state.depth_compare_operator = RD::COMPARE_OP_NEVER;
+				break;
+			default:
+				depth_stencil_state.depth_compare_operator = RD::COMPARE_OP_LESS_OR_EQUAL;
+				break;
+		}
 	}
 	bool depth_pre_pass_enabled = bool(GLOBAL_GET("rendering/driver/depth_prepass/enable"));
```

Comment:
Just want to point out that for opaque materials #70214 will overwrite the compare op you just set later on.
This is currently not a problem due to #73158 (which sends custom no-depth draw materials into the alpha pass) but could be a potential problem in the future.

Reply:
As things are now, it would be difficult to integrate depth functions into the opaque render pass anyways, due to potential sorting issues. I'm currently looking into solutions for this, but it might be beyond the scope of this PR.